src/transformers/hf_logging.py

```diff
@@ -0,0 +1,67 @@
+# coding=utf-8
+# Copyright 2020 Optuna, Hugging Face
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Logging utilities. """
+
+import logging
+import threading
+from typing import Optional
+
+
+from logging import CRITICAL  # NOQA
+from logging import DEBUG  # NOQA
+from logging import ERROR  # NOQA
+from logging import FATAL  # NOQA
+from logging import INFO  # NOQA
+from logging import WARN  # NOQA
+from logging import WARNING  # NOQA
+from logging import NOTSET  # NOQA
+
+_lock = threading.Lock()
+_default_handler: Optional[logging.Handler] = None
+
+
+def _get_library_logger() -> logging.Logger:
```

Comment:
As mentioned above, can we still accept a `name` argument here and set the logger formatting to display it in addition to the name `transformers`?

I think we will probably need to follow more the approach of optuna (have a root logger etc)